pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -9,10 +9,17 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const KUBEVIRT_PROMETHEUS_RULE_NAME = "prometheus-kubevirt-rules"
-const prometheusLabelKey = "prometheus.kubevirt.io"
-const prometheusLabelValue = "true"
-const runbookUrlBasePath = "https://kubevirt.io/monitoring/runbooks/"
+const (
+	KUBEVIRT_PROMETHEUS_RULE_NAME = "prometheus-kubevirt-rules"
+	prometheusLabelKey            = "prometheus.kubevirt.io"
+	prometheusLabelValue          = "true"
+	runbookUrlBasePath            = "https://kubevirt.io/monitoring/runbooks/"
+	severityAlertLabelKey         = "severity"
+	partOfAlertLabelKey           = "kubernetes_operator_part_of"
+	partOfAlertLabelValue         = "kubevirt"
+	componentAlertLabelKey        = "kubernetes_operator_component"
+	componentAlertLabelValue      = "kubevirt"
```

Comment:
We already have constants defining common labels such as `app.kubernetes.io/part-of` and `app.kubernetes.io/component`, so no need to redefine them. Take a look [here](https://github.com/kubevirt/api/blob/331dc8962ac5d69853d81463b4f9d09e7dba4051/core/v1/types.go#L699-L704).

As for the label values, we can follow the same pattern used when generating kubevirt deployments/deamonsets. Take a look [here](https://github.com/kubevirt/kubevirt/blob/7e3d50f91183e7ae03d2e0d7a71f5881520d4d29/pkg/virt-operator/resource/generate/install/strategy.go#L433-L435) and follow the usage of `productName` and `productComponent`.

Reply:
@zcahana Hi, We can't use the label names as they are in [types.go] (https://github.com/kubevirt/api/blob/331dc8962ac5d69853d81463b4f9d09e7dba4051/core/v1/types.go#L699-L704)
since due to the Prometheus naming restrictions. We agreed on this labels naming across all CNV sub operators.

About the values for these keys, Is the app.kubernetes.io/part-of and app.kubernetes.io/component values "kubevirt"?

Reply:
Got it about naming restrictions. It is what it is then. If the name is still open for change, maybe use `app_kubernetes_io_part_of` and `app_kubernetes_io_component` to better hint that this is a transformation of that sort-of standard kubernetes label.

> About the values for these keys, Is the app.kubernetes.io/part-of and app.kubernetes.io/component values "kubevirt"?

Well, not exactly.
- `app.kubernetes.io/component` takes its value from `KubeVirt.Spec.ProductName`, if specified, and otherwise defaults to `kubevirt` ([ref](https://github.com/kubevirt/kubevirt/blob/3ef8b9d61c6ddfc723dbcb09dfda25f9d05e1c9b/pkg/virt-operator/resource/apply/reconcile.go#L96)).
- app.kubernetes.io/part-of takes its value from `KubeVirt.Spec.ProductName`, if specified, and otherwise the label is dropped.

Note that this defaulting mechanism is only applied in `injectOperatorMetadata` upon the resource, just before it's created/updated to the API server. It doesn't operate on nested fields such as `.Spec.Groups[].Rules[].Labels`. Also, this default value doesn't seem to propagate to the [`config`](https://github.com/kubevirt/kubevirt/blob/7e3d50f91183e7ae03d2e0d7a71f5881520d4d29/pkg/virt-operator/resource/generate/install/strategy.go#L440) object which is used create the `productName` and `productComponent` that I referenced above.

Reply:
> Since this will be used to filter the alerts I think it may be a problem to rely on the existing labels, since we need to make sure there is consistency and the naming for "part-of" is the same across all of HCO components

@sradco not sure I understand this comment. Did you mean that you want the label values to be `kubevirt` so that they remain consistent with alerts from other components which also use `kubevirt` as the name?

As it's implemented now, we might have an inconsistency between the prometheus labels (`kubernetes_operator_part_of`, ...) and the kubernetes labels (`app.kubernetes.io/part-of`, ...) if their values are overriden via the KubeVirt CR. I presume this maybe the case in downstream CNV builds or other products built atop of kubevirt.

Reply:
@zcahana Yes, the "kubevirt" value will be the same for all hco components so we can filter the alerts based on it. And the component will mean the sub operator name.
app.kubernetes.io/part-of is translated in prometheus to app_kubernetes_io_part_of. This is the reason we use a different label kubernetes_operator_part_of so they will not collide.

Reply:
But wouldn't it be more consistent if the values of both of these will match? As is now, we might have `app_kubernetes_io_part_of=kubevirt` and `app.kubernetes.io/part-of=cnv` (just an example, I don't know what's being used downstream).

Reply:
@zcahana Its similar but its kubernetes_operator_part_of not app_kubernetes_io_part_of

Reply:
Oh yeah, I meant `kubernetes_operator_part_of=kubevirt` (copy paste error). Anyway I think the point is clear. You explicitly distinguish between these two labels and don't mind them having different values.